Adding module for gobblin-cluster

abti opened the pull request:
Adding module for gobblin-cluster that will: 
- Contain code to help run Gobblin in standalone static or elastic cluster, as well as 
- Act as abstraction and base implementation for Gobblin on Yarn, Mesos, etc.

abti commented:
@shirshanka Please review. This is a skeleton module with few prerequisite and redundant dependencies that we can clean up as we start writing code

shirshanka commented on gobblin-cluster/build.gradle:
```diff
@@ -0,0 +1,99 @@
+  compile externalDependency.log4j
+  compile externalDependency.commonsCli
+  compile externalDependency.gson
+  compile externalDependency.hiveCommon
```
Are the dependencies a copy from another module?

abti replied:
yes, from gobblin-yarn, so we also have a few Yarn related dependencies (which would be removed as we develop, since this should not eventually depend on Yarn)

abti committed:
Adding module for gobblin-cluster that will:

- Contain code to help run Gobblin in standalone static or elastic cluster, as well as
- Act as abstraction and base implementation for Gobblin on Yarn, Mesos, etc.

abti commented:
Incorporated review feedback. Cleaned up build.gradle a bit to limit dependencies to only what we might need

shirshanka commented:
LGTM
